app/models/user.rb

```diff
@@ -167,12 +176,92 @@ def self.requested_sync_approval
+        when :Facility
+          organizations = Organization.includes(:facilities).where(facilities: scope)
+          facility_group = scope.facility_group
+
+          accesses
+            .where(role: admin, resourceable: organizations)
+            .or(accesses.where(role: admin, resourceable: facility_group))
+            .or(accesses.where(role: admin, resourceable: scope)).exists?
+        else
+          nil
+      end
+
+    return value unless value.nil?
+
+    case scope.class.to_sym
+      when Class
+        case scope.model_name.name.to_sym
+          when :Organization
+            accesses.where(role: admin, resourceable_type: %w[Organization]).exists?
+          when :FacilityGroup
+            accesses.where(role: admin, resourceable_type: %w[Organization FacilityGroup]).exists?
+          when :Facility
+            accesses.where(role: admin, resourceable_type: %w[Organization FacilityGroup Facility]).exists?
+          else
+            raise ArgumentError, "Invalid scope name"
+        end
+      when ActiveRecord::Relation
+        case scope.model_name.name.to_sym
+          when :Organization
+            accesses.where(role: admin, resourceable: scope).exists?
+          when :FacilityGroup
+            organizations = Organization.includes(:facility_groups).where(facility_groups: scope)
+
+            accesses
+              .where(role: admin, resourceable: organizations)
+              .or(accesses.where(role: admin, resourceable: scope)).exists?
+
+          when :Facility
+            organizations = Organization.includes(:facilities).where(facilities: scope)
+            facility_group = scope.facility_group
+
+            accesses
+              .where(role: admin, resourceable: organizations)
+              .or(accesses.where(role: admin, resourceable: facility_group))
+              .or(accesses.where(role: admin, resourceable: scope)).exists?
+
+          else
+            raise ArgumentError, "Invalid scope name"
+        end
+      else
+        raise ArgumentError, "Invalid scope type"
+    end
   end
 
-  def resources
-    user_permissions.map(&:resource)
+  def read_agg_facilities
+    read_agg_organizations =
+      Facility.where(id: accesses
+                           .where(role: [Role.admin.first, Role.analyst.first], resourceable_type: "Organization")
+                           .map(&:resourceable)
+                           .map(&:facilities))
+    read_agg_facility_groups =
+      Facility.where(id: accesses
+                           .where(role: [Role.admin.first, Role.analyst.first], resourceable_type: "FacilityGroup")
+                           .map(&:resourceable)
+                           .map(&:facilities))
+    read_agg_facilities =
+      Facility.where(id: accesses
+                           .where(role: [Role.admin.first, Role.analyst.first], resourceable_type: "Facility")
+                           .map(&:resourceable))
```

Comment:
This can now collapse to

```
Facility.where(id: access
  .where(role: [Role.admin.first, Role.analyst.first])
  .map(&:resourceable)
  .map(&:facilities) 
)
```
because all three models return sensible results for `object.facilities`